cmd/juju/commands/upgradecontroller.go

```diff
@@ -98,11 +98,10 @@ func (c *upgradeControllerCommand) Run(ctx *cmd.Context) (err error) {
 
 func (c *upgradeControllerCommand) upgradeController(ctx *cmd.Context, controllerModel string) error {
 	jcmd := &upgradeJujuCommand{
-		baseUpgradeCommand: baseUpgradeCommand{
-			upgradeMessage: "upgrade to this version by running\n    juju upgrade-controller",
-		},
-		jujuClientAPI: c.jujuClientAPI,
+		baseUpgradeCommand: c.baseUpgradeCommand,
+		jujuClientAPI:      c.jujuClientAPI,
 	}
+	jcmd.upgradeMessage = "upgrade to this version by running\n    juju upgrade-controller"
```

Comment:
move up to the struct initialisation

Reply:
This change fixed a previous bug.
We need to pass the `c.baseUpgradeCommand` rather than a new `baseUpgradeCommand{}`.
Otherwise, `c.baseUpgradeCommand.SetFlags(f)` in `upgradeControllerCommand.SetFlags` gets lost.